crates/rooch-rpc-api/src/jsonrpc_types/address.rs

```diff
@@ -63,69 +67,101 @@ impl From<RoochAddressView> for AccountAddress {
 }
 
 #[derive(Debug, Clone)]
-pub struct RoochOrBitcoinAddress {
+pub struct UnitedAddress {
     pub rooch_address: RoochAddress,
     pub bitcoin_address: Option<BitcoinAddress>,
+    pub nostr_public_key: Option<NostrPublicKey>,
 }
 
-pub type RoochOrBitcoinAddressView = StrView<RoochOrBitcoinAddress>;
+pub type UnitedAddressView = StrView<UnitedAddress>;
 
-impl std::fmt::Display for RoochOrBitcoinAddressView {
+impl std::fmt::Display for UnitedAddressView {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        if self.0.bitcoin_address.is_some() {
-            return write!(f, "{}", self.0.bitcoin_address.as_ref().unwrap());
+        if let Some(bitcoin_address) = &self.0.bitcoin_address {
+            return write!(f, "{}", bitcoin_address);
+        }
+        if let Some(nostr_public_key) = &self.0.nostr_public_key {
+            return write!(f, "{}", nostr_public_key);
         }
         write!(f, "{}", self.0.rooch_address)
     }
 }
 
-impl FromStr for RoochOrBitcoinAddressView {
+impl FromStr for UnitedAddressView {
     type Err = anyhow::Error;
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match RoochAddress::from_str(s) {
-            Ok(rooch_address) => Ok(StrView(RoochOrBitcoinAddress {
+            Ok(rooch_address) => Ok(StrView(UnitedAddress {
                 rooch_address,
                 bitcoin_address: None,
+                nostr_public_key: None,
             })),
-            Err(_) => {
-                let bitcoin_address = BitcoinAddress::from_str(s)?;
-                Ok(StrView(RoochOrBitcoinAddress {
-                    rooch_address: bitcoin_address.to_rooch_address(),
-                    bitcoin_address: Some(bitcoin_address),
-                }))
-            }
+            Err(_) => match XOnlyPublicKey::from_bech32(s) {
+                Ok(x_only_pk) => {
+                    let nostr_public_key = NostrPublicKey::new(x_only_pk);
+                    let bitcoin_address =
+                        nostr_public_key.to_bitcoin_address(Network::Bitcoin.to_num())?; // TODO: use network variable?
```

Comment:
Rooch BitcoinAddress ignores the network, so using `Network::Bitcoin.to_num()` is ok.

Reply:
Resolved.